imod/tests/test_mf6/test_mf6_regrid_simulation.py

```diff
@@ -51,3 +53,26 @@ def test_regridded_simulation_has_required_packages(
         new_simulation["time_discretization"], imod.mf6.TimeDiscretization
     )
     assert isinstance(new_simulation["flow"], imod.mf6.GroundwaterFlowModel)
+
+
+@pytest.mark.usefixtures("circle_model")
+def test_regrid_with_methods_without_functions(circle_model, tmp_path):
+    simulation = circle_model
+    idomain = circle_model["GWF_1"].domain
+    # redefine the default regridding method for the constant head package,
+    # assign a default method that does not have a function
+    old_regrid_method = imod.mf6.ConstantHead._regrid_method
+    imod.mf6.ConstantHead._regrid_method = {
+        "head": (RegridderType.BARYCENTRIC,),
+        "concentration": (RegridderType.BARYCENTRIC,),
+    }
+    regridding_succeeded = False
+
+    # try regridding the simulation with the new default method
+    try:
+        simulation.regrid_like("regridded", idomain)
+        regridding_succeeded = True
+    finally:
+        imod.mf6.ConstantHead._regrid_method = old_regrid_method
```

Comment:
The scope of circle_model is "function", but we still need to put the default regridding method back, because it's a class variable and not an instance variable.